e2e-tests/insurance/cypress/e2e/journeys/declarations/modern-slavery/modern-slavery-expandable-definition-content.spec.js

```diff
@@ -0,0 +1,199 @@
+const assertTermColumn = (selector, content) => {
+  cy.checkText(selector, content.TERM);
+};
+
+const baseUrl = Cypress.config('baseUrl');
+
+context('Insurance - Declarations - Modern slavery page - expandable `definition` content', () => {
+  let referenceNumber;
+  let url;
+
+  before(() => {
+    cy.completeSignInAndGoToApplication({}).then(({ referenceNumber: refNumber }) => {
+      referenceNumber = refNumber;
+
+      url = `${baseUrl}${INSURANCE_ROOT}/${referenceNumber}${MODERN_SLAVERY}`;
+
+      cy.navigateToUrl(url);
+      cy.assertUrl(url);
+    });
+  });
+
+  beforeEach(() => {
+    cy.saveSession();
+
+    cy.navigateToUrl(url);
+  });
+
+  after(() => {
+    cy.deleteApplication(referenceNumber);
+  });
+
+  it('renders summary text', () => {
+    expandable.summary().should('exist');
+
+    cy.checkText(expandable.summary(), INTRO);
+  });
+
+  it('clicking summary text reveals details', () => {
+    expandable.summary().click();
+
+    expandable.table.headers.term().should('be.visible');
+    expandable.table.headers.definition().should('be.visible');
+  });
+
+  describe('table', () => {
+    const { BODY, HEADERS } = TABLE;
+
+    let row;
+    let content;
+
+    it('renders table headers', () => {
+      expandable.summary().click();
+
+      cy.checkText(expandable.table.headers.term(), HEADERS.TERM);
+      cy.checkText(expandable.table.headers.definition(), HEADERS.DEFINITION);
+    });
+
+    describe('body row 1', () => {
+      beforeEach(() => {
+        expandable.summary().click();
+
+        row = expandable.table.body.row(1);
+
+        const { 0: contentRow } = BODY;
+        content = contentRow;
+      });
+
+      it('renders `term` column', () => {
+        assertTermColumn(row.term(), content);
+      });
+
+      it('renders `definition` column', () => {
+        cy.checkText(row.definition(), content.DEFINITION[0].TEXT);
+      });
+    });
+
+    describe('body row 2', () => {
+      beforeEach(() => {
+        expandable.summary().click();
+
+        row = expandable.table.body.row(2);
+
+        const { 1: contentRow } = BODY;
+        content = contentRow;
+      });
+
+      it('renders `term` column', () => {
+        assertTermColumn(row.term(), content);
+      });
+
+      it('renders `definition` column', () => {
+        cy.checkText(row.definition(), content.DEFINITION[0].TEXT);
+      });
+    });
+
+    describe('body row 3', () => {
+      beforeEach(() => {
+        expandable.summary().click();
+
+        row = expandable.table.body.row(3);
+
+        const { 2: contentRow } = BODY;
+        content = contentRow;
+      });
+
+      it('renders `term` column', () => {
+        assertTermColumn(row.term(), content);
+      });
+
+      it('renders `definition` column', () => {
+        row
+          .definitionListItemChild(1, 1)
+          .invoke('text')
+          .then((text) => {
+            expect(text).includes(content.DEFINITION[0].CHILDREN[0].TEXT);
```

Comment:
would it be better to put content.DEFINITION[0] in a const?

Reply:
I think it is better to keep as is so that it has the full context in a single line.

These tests are already quite complex/fiddly and very index based - I think an additional const and line of code could confuse things / make it harder to read and debug in the future.